src/main/kotlin/tech/thatgravyboat/craftify/ui/UIControls.kt

```diff
@@ -122,39 +115,47 @@ class UIControls : UIContainer() {
-            shuffleButton.constraints.x = 42.pixels() - 24.pixels()
-            prevButton.constraints.x = 42.pixels() - 12.pixels()
-            playButton.constraints.x = 42.pixels()
-            nextButton.constraints.x = 42.pixels() + 12.pixels()
-            repeatButton.constraints.x = 42.pixels() + 24.pixels()
-            externalButton.constraints.x = 42.pixels() + 36.pixels()
+
+        var activeButtons = updateButton(settingsButton, ThemeConfig.showSettingsButton, ThemeConfig.settingsIcon, settings)
+        activeButtons = updateButton(shuffleButton, ThemeConfig.showShuffleButton, ThemeConfig.shuffleIcon, shuffle, activeButtons)
+        activeButtons = updateButton(prevButton, ThemeConfig.showPreviousButton, ThemeConfig.previousIcon, prev, activeButtons)
+        activeButtons = updateButton(nextButton, ThemeConfig.showNextButton, ThemeConfig.nextIcon, next, activeButtons)
+        activeButtons = updateButton(repeatButton, ThemeConfig.showRepeatButton, ThemeConfig.repeatIcon, repeat, activeButtons)
+        activeButtons = updateButton(externalButton, ThemeConfig.showExternalButton, ThemeConfig.externalIcon, external, activeButtons)
+
+        if (ThemeConfig.showPlayButton) {
+            playButton.updateImage(notNullNotBlankOrElse(ThemeConfig.pauseIcon, pause), notNullNotBlankOrElse(ThemeConfig.playIcon, play))
+            playButton.unhide()
+            activeButtons++
         } else {
-            this.addChild(positionButton)
-            this.addChild(volumeButton)
-            positionButton.constraints.x = 65.pixels() - 48.pixels()
-            settingsButton.constraints.x = 65.pixels() - 36.pixels()
-            shuffleButton.constraints.x = 65.pixels() - 24.pixels()
-            prevButton.constraints.x = 65.pixels() - 12.pixels()
-            playButton.constraints.x = 65.pixels()
-            nextButton.constraints.x = 65.pixels() + 12.pixels()
-            repeatButton.constraints.x = 65.pixels() + 24.pixels()
-            externalButton.constraints.x = 65.pixels() + 36.pixels()
-            volumeButton.constraints.x = 65.pixels() + 48.pixels()
+            playButton.hide(true)
         }
+
+        if(ThemeConfig.showVolumeButton && (!ThemeConfig.hideImage || activeButtons < 7)) {
+            volumeButton.updateImage(notNullNotBlankOrElse(ThemeConfig.volumeIcon, volume))
+            volumeButton.unhide()
+            activeButtons++
+        } else {
+            volumeButton.hide(true)
+        }
+
+        if (ThemeConfig.showPositionEditorButton && (!ThemeConfig.hideImage || activeButtons < 7)) {
+            positionButton.updateImage(notNullNotBlankOrElse(ThemeConfig.positionEditorIcon, position))
+            positionButton.unhide()
+        } else  {
+            positionButton.hide(true)
+        }
+    }
+
+    private fun updateButton(button: UIButton, visible: Boolean, image: String, default: String, activeButtons: Int = 0): Int {
```

Comment:
change to allow secondary optional image and use for all buttons including volume, position, and play to keep consistency.